File: python3/1679.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        s = 0
        e = len(nums) - 1
        count = 0
        while s < e:
            t = nums[s] + nums[e]
            if t == k:
                count += 1
                s += 1
                e -= 1
            elif t < k:
                s += 1
            else:
                e -= 1
        return count

File: java/1679.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        Arrays.sort(nums);
        int s = 0, e = nums.length - 1, count = 0;
        while (s < e) {
            int t = nums[s] + nums[e];
            if (t == k) {
                count++;
                s++;
                e--;
            } else if (t < k)
                s++;
            else
                e--;
        }
        return count;
    }
}
